Error while checking build/O2/fullCI for dc4f7c013ba3db8adbf0453ceee33d8b1f3051ed at 2023-03-28 02:49:
```
## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile reco_NOGPU.log
reco_NOGPU.log-[FATAL] Illegal command-line key/value string: ITSVertexerParam.nThreads=
reco_NOGPU.log:[FATAL] unknown error while setting up workflow in o2-its-reco-workflow.
reco_NOGPU.log-[ERROR] invalid workflow in o2-ft0-reco-workflow: No matching output found for FT0/DIGITSBC/0 as requested by data processor "ft0-reconstructor". Candidates:
reco_NOGPU.log-[ERROR] -CTP/OrbitReset/0
[FATAL] Illegal command-line key/value string: ITSVertexerParam.nThreads=
[FATAL] unknown error while setting up workflow in o2-its-reco-workflow.
[ERROR] invalid workflow in o2-ft0-reco-workflow: No matching output found for FT0/DIGITSBC/0 as requested by data processor "ft0-reconstructor". Candidates:
[ERROR] -CTP/OrbitReset/0
[ERROR] -FT0/TimeSpectraInfo/0
[ERROR] -FT0/RECCHDATA/0
[ERROR] -FT0/RECPOINTS/0
[ERROR] 


## sw/BUILD/o2checkcode-latest/log
--
++ echo '========== List of errors found =========='
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/9bda2414f61cc14ce7b82d1493265848eb5f12a2/slc8_x86-64/o2checkcode/1.0-local297/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11029/dc4f7c013ba3db8adbf0453ceee33d8b1f3051ed/build_O2_fullCI/pretty.html).

--- a/Detectors/FIT/FDD/dcsmonitoring/workflow/FDDDCSConfigProcessorSpec.h
+++ b/Detectors/FIT/FDD/dcsmonitoring/workflow/FDDDCSConfigProcessorSpec.h
@@ -45,7 +45,9 @@ DataProcessorSpec getFDDDCSConfigProcessorSpec()
     outputs,
     AlgorithmSpec{adaptFromTask<o2::fit::FITDCSConfigProcessor>("FDD", ddDChM)},
     Options{{"use-verbose-mode", VariantType::Bool, false, {"Use verbose mode"}},
-            {"filename-dchm", VariantType::String, "FDD-deadchannels.txt", {"Dead channel map file name"}}}};
+            {"filename-dchm", VariantType::String, "FDD-deadchannels.txt", {"Dead channel map file name"}},
+            {"valid-days-dchm", VariantType::UInt32, 180u, {"Dead channel map validity in days"}},
+            {"no-validate", VariantType::Bool, false, {"Don't validate the CCDB uploads"}}}};
 }
 
 } // namespace framework

--- a/Detectors/FIT/FT0/dcsmonitoring/workflow/FT0DCSConfigProcessorSpec.h
+++ b/Detectors/FIT/FT0/dcsmonitoring/workflow/FT0DCSConfigProcessorSpec.h
@@ -64,7 +64,9 @@ DataProcessorSpec getFT0DCSConfigProcessorSpec()
     outputs,
     AlgorithmSpec{adaptFromTask<o2::ft0::FT0DCSConfigProcessor>("FT0", ddDChM)},
     Options{{"use-verbose-mode", VariantType::Bool, false, {"Use verbose mode"}},
-            {"filename-dchm", VariantType::String, "FT0-deadchannels.txt", {"Dead channel map file name"}}}};
+            {"filename-dchm", VariantType::String, "FT0-deadchannels.txt", {"Dead channel map file name"}},
+            {"valid-days-dchm", VariantType::UInt32, 180u, {"Dead channel map validity in days"}},
+            {"no-validate", VariantType::Bool, false, {"Don't validate the CCDB uploads"}}}};
 }
 
 } // namespace framework

--- a/Detectors/FIT/FV0/dcsmonitoring/workflow/FV0DCSConfigProcessorSpec.h
+++ b/Detectors/FIT/FV0/dcsmonitoring/workflow/FV0DCSConfigProcessorSpec.h
@@ -45,7 +45,9 @@ DataProcessorSpec getFV0DCSConfigProcessorSpec()
     outputs,
     AlgorithmSpec{adaptFromTask<o2::fit::FITDCSConfigProcessor>("FV0", ddDChM)},
     Options{{"use-verbose-mode", VariantType::Bool, false, {"Use verbose mode"}},
-            {"filename-dchm", VariantType::String, "FV0-deadchannels.txt", {"Dead channel map file name"}}}};
+            {"filename-dchm", VariantType::String, "FV0-deadchannels.txt", {"Dead channel map file name"}},
+            {"valid-days-dchm", VariantType::UInt32, 180u, {"Dead channel map validity in days"}},
+            {"no-validate", VariantType::Bool, false, {"Don't validate the CCDB uploads"}}}};
 }
 
 } // namespace framework

--- a/Detectors/FIT/common/dcsmonitoring/include/FITDCSMonitoring/FITDCSConfigProcessorSpec.h
+++ b/Detectors/FIT/common/dcsmonitoring/include/FITDCSMonitoring/FITDCSConfigProcessorSpec.h
@@ -48,11 +48,17 @@ class FITDCSConfigProcessor : public o2::framework::Task
   {
     initDCSConfigReader();
     mDCSConfigReader->setFileNameDChM(ic.options().get<string>("filename-dchm"));
+    mDCSConfigReader->setValidDaysDChM(ic.options().get<uint>("valid-days-dchm"));
     mDCSConfigReader->setCcdbPathDChM(mDetectorName + "/Calib/DeadChannelMap");
     mVerbose = ic.options().get<bool>("use-verbose-mode");
     mDCSConfigReader->setVerboseMode(mVerbose);
+    mValidateUpload = !ic.options().get<bool>("no-validate");
+    mDCSConfigReader->setValidateUploadMode(mValidateUpload);
+
     LOG(info) << "Verbose mode: " << mVerbose;
+    LOG(info) << "Validate upload: " << mValidateUpload;
     LOG(info) << "Expected dead channel map file name: " << mDCSConfigReader->getFileNameDChM();
+    LOG(info) << "Dead channel maps will be valid for " << mDCSConfigReader->getValidDaysDChM() << " days";
   }
 
   void run(o2::framework::ProcessingContext& pc) final
@@ -123,6 +129,7 @@ class FITDCSConfigProcessor : public o2::framework::Task
   std::string mDetectorName;                        ///< Detector name
   o2::header::DataDescription mDataDescriptionDChM; ///< DataDescription for the dead channel map
   bool mVerbose = false;                            ///< Verbose mode
+  bool mValidateUpload = true;                      ///< Validate upload
 };
 
 } // namespace fit

--- a/Detectors/FIT/common/dcsmonitoring/include/FITDCSMonitoring/FITDCSConfigReader.h
+++ b/Detectors/FIT/common/dcsmonitoring/include/FITDCSMonitoring/FITDCSConfigReader.h
@@ -52,8 +52,14 @@ class FITDCSConfigReader
   const std::string& getFileNameDChM() const;
   void setFileNameDChM(const std::string& fileName);
 
+  const uint getValidDaysDChM() const;
+  void setValidDaysDChM(const uint validDays);
+
   const bool getVerboseMode() const;
   void setVerboseMode(const bool verboseMode);
+
+  const bool getValidateUploadMode() const;
+  void setValidateUploadMode(const bool validateUpload);
 
  protected:
   o2::fit::DeadChannelMap mDChM; ///< The dead channel map CCDB object
@@ -61,9 +67,11 @@ class FITDCSConfigReader
 
  private:
   std::string mFileNameDChM;                                              ///< The expected file name of the dead channel map
+  uint mValidDaysDChM = 180u;                                             ///< The dead channel map validity in days
   std::string mCcdbPathDChM;                                              ///< The dead channel map CCDB path
   long mStartValidityDChM = o2::ccdb::CcdbObjectInfo::INFINITE_TIMESTAMP; ///< Start validity of the dead channel map CCDB object
   o2::ccdb::CcdbObjectInfo mCcdbObjectInfoDChM;                           ///< CCDB object info for the dead channel map
+  bool mValidateUpload = true;                                            ///< Validate upload mode
 
   ClassDefNV(FITDCSConfigReader, 1);
 };

--- a/Detectors/FIT/common/dcsmonitoring/src/FITDCSConfigReader.cxx
+++ b/Detectors/FIT/common/dcsmonitoring/src/FITDCSConfigReader.cxx
@@ -47,6 +47,7 @@ void FITDCSConfigReader::updateDChMCcdbObjectInfo()
 {
   std::map<std::string, std::string> metadata;
   o2::calibration::Utils::prepareCCDBobjectInfo(mDChM, mCcdbObjectInfoDChM, mCcdbPathDChM, metadata, getStartValidityDChM(), getEndValidityDChM());
+  mCcdbObjectInfoDChM.setValidateUpload(getValidateUploadMode());
 }
 
 const o2::fit::DeadChannelMap& FITDCSConfigReader::getDChM() const { return mDChM; }
@@ -54,7 +55,7 @@ void FITDCSConfigReader::resetDChM() { mDChM.clear(); }
 const std::string& FITDCSConfigReader::getCcdbPathDChm() const { return mCcdbPathDChM; }
 void FITDCSConfigReader::setCcdbPathDChM(const std::string& ccdbPath) { mCcdbPathDChM = ccdbPath; }
 const long FITDCSConfigReader::getStartValidityDChM() const { return mStartValidityDChM; }
-const long FITDCSConfigReader::getEndValidityDChM() const { return mStartValidityDChM + o2::ccdb::CcdbObjectInfo::MONTH; }
+const long FITDCSConfigReader::getEndValidityDChM() const { return mStartValidityDChM + getValidDaysDChM() * o2::ccdb::CcdbObjectInfo::DAY; }
 void FITDCSConfigReader::setStartValidityDChM(const long startValidity) { mStartValidityDChM = startValidity; }
 const bool FITDCSConfigReader::isStartValidityDChMSet() const { return mStartValidityDChM != o2::ccdb::CcdbObjectInfo::INFINITE_TIMESTAMP; }
 void FITDCSConfigReader::resetStartValidityDChM() { mStartValidityDChM = o2::ccdb::CcdbObjectInfo::INFINITE_TIMESTAMP; }
@@ -64,5 +65,11 @@ o2::ccdb::CcdbObjectInfo& FITDCSConfigReader::getObjectInfoDChM() { return mCcdb
 const std::string& FITDCSConfigReader::getFileNameDChM() const { return mFileNameDChM; }
 void FITDCSConfigReader::setFileNameDChM(const std::string& fileName) { mFileNameDChM = fileName; }
 
+const uint FITDCSConfigReader::getValidDaysDChM() const { return mValidDaysDChM; }
+void FITDCSConfigReader::setValidDaysDChM(const uint validDays) { mValidDaysDChM = validDays; }
+
 const bool FITDCSConfigReader::getVerboseMode() const { return mVerbose; }
 void FITDCSConfigReader::setVerboseMode(const bool verboseMode) { mVerbose = verboseMode; }
+
+const bool FITDCSConfigReader::getValidateUploadMode() const { return mValidateUpload; }
+void FITDCSConfigReader::setValidateUploadMode(const bool validateUpload) { mValidateUpload = validateUpload; };

